diff --git a/README.md b/README.md
index f31cdf9..a3c0131 100644
--- a/README.md
+++ b/README.md
@@ -4,6 +4,6 @@ # OCHA Snap Service integration for Drupal.
 
 # Quick Start
 
-Install this module and add a function that calls `ocha_snap($url)`.
+Install this module and add a function that calls `ocha_snap_generate($url)`.
 
 All Snap Service parameters are documented at https://github.com/UN-OCHA/tools-snap-service#parameters
diff --git a/ocha_snap.module b/ocha_snap.module
index dc4a5cc..2001ad7 100644
--- a/ocha_snap.module
+++ b/ocha_snap.module
@@ -121,7 +121,13 @@ function ocha_snap_generate($url, array $params = []) {
     $output = chr_curl_http_request($url, $req_params);
   }
   else {
-    $output = \Drupal::httpClient()->post($url, $req_params);
+    try {
+      $output = \Drupal::httpClient()->post($url, $req_params);
+    }
+    catch (Exception $e) {
+      \Drupal::logger('ocha_snap')->error('%e Exception: Failed to get a response for a %output for "%u".', ['%e' => get_class($e), '%output' => $params['output'], '%u' => $params['url']]);
+      return FALSE;
+    }
   }
 
   // Basic error checking.
