diff --git a/core/lib/Drupal/Core/Render/RenderCache.php b/core/lib/Drupal/Core/Render/RenderCache.php
index ed7771b8047..53c547d59a0 100644
--- a/core/lib/Drupal/Core/Render/RenderCache.php
+++ b/core/lib/Drupal/Core/Render/RenderCache.php
@@ -127,7 +127,7 @@ public function set(array &$elements, array $pre_bubbling_elements) {
    */
   public function getCacheableRenderArray(array $elements) {
     $data = [
-      '#markup' => $elements['#markup'],
+      '#markup' => isset($elements['#markup']) ? $elements['#markup'] : '',
       '#attached' => $elements['#attached'],
       '#cache' => [
         'contexts' => $elements['#cache']['contexts'],
