diff --git a/core/lib/Drupal/Core/Render/RenderCache.php b/core/lib/Drupal/Core/Render/RenderCache.php
index 674c00e28f..d46e6f0f6d 100644
--- a/core/lib/Drupal/Core/Render/RenderCache.php
+++ b/core/lib/Drupal/Core/Render/RenderCache.php
@@ -330,7 +330,7 @@ protected function createCacheID(array &$elements) {
   public function getCacheableRenderArray(array $elements) {
     $data = [
-      '#markup' => $elements['#markup'],
+      '#markup' => isset($elements['#markup']) ? $elements['#markup'] : '',
       '#attached' => $elements['#attached'],
       '#cache' => [
         'contexts' => $elements['#cache']['contexts'],
