diff --git a/src/Service/BlockClassHelperService.php b/src/Service/BlockClassHelperService.php
index d8dbf80a6b490c8f12f2d215d8361242450b480e..f920bcf009286695b4f08700ef1d05b4b6ffa71d 100644
--- a/src/Service/BlockClassHelperService.php
+++ b/src/Service/BlockClassHelperService.php
@@ -193,7 +193,7 @@ class BlockClassHelperService {
       $config = $this->configFactory->getEditable('block_class.settings');
 
       // Get the current classes stored.
-      $block_classes_stored = $config->get('block_classes_stored');
+      $block_classes_stored = $config->get('block_classes_stored') ?? [];
 
       // Get the current class and export to array.
       $current_block_classes = explode(' ', $block_classes ?? '');
